user/tests/wiring/power_management_long_running/power.cpp

```diff
@@ -0,0 +1,384 @@
+
+        SERIAL.println("> Please \"unplug\" the USB");
+        while (Serial.isConnected());
+
+        SERIAL.println("> Wait the power management settled...");
+        assertTrue(waitFor([]{
+            return System.batteryState() == BATTERY_STATE_DISCHARGING;
+        }, 10000));
+        assertEqual(System.powerSource(), (int)POWER_SOURCE_BATTERY);
+
+        SERIAL.print("> Device is connecting to the cloud... ");
+        Particle.connect();
+        assertTrue(waitFor(Particle.connected, 120000)); // 2 minutes
+        SERIAL.println("connected.");
+
+        SERIAL.println("> Device is publishing message in every second to discharge the battery...");
+        FuelGauge fuel;
+        float vcell = fuel.getVCell();
+        SERIAL.printlnf("> VBAT: %0.2fv", vcell);
+
+        SERIAL.println("> Test is running...");
+        uint8_t count = 0;
+        time32_t now = Time.now();
+        while (vcell > DISCHARGE_TERMINATION_VCELL || count < 10) {
+            if (Time.now() - now > 1) {
+                now = Time.now();
+                Particle.publish("Hello world!");
+                assertEqual(System.batteryState(), (int)BATTERY_STATE_DISCHARGING);
+                assertEqual(System.powerSource(), (int)POWER_SOURCE_BATTERY);
+
+                vcell = fuel.getVCell();
+                if (vcell <= DISCHARGE_TERMINATION_VCELL) {
+                    count++;
+                }
+            }
+        }
+    }
+}
+
+} // anonymous
+
+test(power_00_setup) {
+    SERIAL.begin(115200);
+    SERIAL.println("\r\npower_00_setup");
+
+    constexpr float DISCHARGE_VCELL = 4.1f;
+
+    auto cfg = System.getPowerConfiguration();
+    assertEqual(cfg.isFeatureSet(SystemPowerFeature::PMIC_DETECTION), true);
+    assertEqual(cfg.isFeatureSet(SystemPowerFeature::USE_VIN_SETTINGS_WITH_USB_HOST), false);
+    assertEqual(cfg.isFeatureSet(SystemPowerFeature::DISABLE), false);
+    assertEqual(cfg.isFeatureSet(SystemPowerFeature::DISABLE_CHARGING), false);
+    assertEqual(cfg.powerSourceMaxCurrent(), particle::power::DEFAULT_INPUT_CURRENT_LIMIT);
+    assertEqual(cfg.powerSourceMinVoltage(), particle::power::DEFAULT_INPUT_VOLTAGE_LIMIT);
+    assertEqual(cfg.batteryChargeCurrent(), particle::power::DEFAULT_CHARGE_CURRENT);
+    assertEqual(cfg.batteryChargeVoltage(), particle::power::DEFAULT_TERMINATION_VOLTAGE);
+
+    while (SERIAL.available() > 0) {
+        (void)SERIAL.read();
+    }
+
+    SERIAL.println("> Please \"plug\" both of the battery and USB. Press any key to continue when the hardware is set up.");
+    while (!Serial.isConnected());
+    while (SERIAL.available() <= 0);
+    while (SERIAL.available() > 0) {
+        (void)SERIAL.read();
+    }
+
+    SERIAL.println("> Wait the power management settled...");
+    assertTrue(waitFor([]{
+        return (System.batteryState()!= BATTERY_STATE_UNKNOWN) && (System.powerSource() == POWER_SOURCE_USB_HOST);
+    }, 10000));
+    PMIC power;
+    assertTrue(power.isChargingEnabled());
```

Comment:
This fails only if the `SystemPowerFeature::DISABLE_CHARGING` is `true` or the battery is disconnected. Otherwise, the PMIC might have tweaked its register for some reason.

Reply:
It's probably that on startup we get continuous charged event even if when we raise the recharge threshold to be 300mV